home/home3.1/src/InverseCalculator.java

```diff
@@ -0,0 +1,14 @@
+public class InverseCalculator {
+    public double inverse(double x, double y, double z) {
+        double product = x * y * z;
+        double sum = x + y + z;
```

Comment:
we can calculate just before usage to optimise the program. in this case it will not be a big different but with bigger entities like images it may be a significant delta
double product =...
if (product ..)
    return ...

double sum = ...
if (sum ...)
     return ...

return ...